crates/stc_ts_file_analyzer/src/analyzer/tsc_helper.rs

```diff
@@ -72,7 +72,13 @@ impl Analyzer<'_, '_> {
         } else if s.starts_with("0b") || s.starts_with("0B") {
             BigInt::parse_bytes(s[2..].as_bytes(), 2)
         } else {
-            s.parse::<BigInt>().ok()
+            // BigInt strings only accepts numbers
+            // 1000n or 1_000n are both considered invalid
+            if s.parse::<i64>().is_ok() {
+                s.parse::<BigInt>().ok()
```

Comment:
I feel like this can be done better (perhaps convert an i64 to BigInt ?) but I couldnt get something working